wpengine/faustjs
# handleSitemapRequests() Faust.js Middleware

## Description

Create a Next.js middleware to handle WordPress sitemaps. The API is as follows:

```tsx
import { handleSitemapRequests } from '@faustjs/next';

handleSitemapRequests(middlewareReq, {
  // Your WordPress URL
  wpUrl: process.env.NEXT_PUBLIC_WORDPRESS_URL,
  // The path to the root sitemap index
  sitemapIndexPath: '/wp-sitemap.xml',
  /**
   * Sitemap paths to ignore. Useful if you don't want to include
   * sitemaps for users, categories, etc.
   */
  sitemapPathsToIgnore: [
    '/wp-sitemap-posts-page-1.xml',
    '/wp-sitemap-posts-post-*', // allows wildcards as well to avoid multiple indices if necessary
  ],
  /**
   * List of Next.js pages to include in your sitemap.
   */
  pages: [
    {
      path: '/about',
      priority: 0.75,
      changefreq: 'monthly',
      lastmod: new Date().toISOString(),
    },
  ],
  /**
   * Replace WP urls with the headless frontend.
   */
  replaceUrls: true,
});

```


## Testing

Test cases have been provided in the PR.


### Manual Testing
1. View `src/pages/_middleware.ts`
2. Notice we are using `/wp-sitemap.xml` as our sitemap index. We're also excluding any sitemaps from the index that start with `/wp-sitemap-users-*`, and we are replacing urls from the WP site to the headless frontend
3. Compare the headless sitemap (http://localhost:3000/wp-sitemap.xml) that is handled by our api, to the WordPress sitemap (http://your-wp-site.com/wp-sitemap.xml)
4. Notice the authors sitemap is not added to the headless sitemap index, and we are also replacing URLs
5. Visit each headless sitemaps in the sitemap index and verify the URLs are correct with proper replacements
6. Add `/wp-sitemap-posts-page-1.xml` to the `sitemapPathsToIgnore` option and notice it no longer shows up in the sitemap index
7. Add `pages: [{path: '/about'}]` to the `handleSitemapRequests` config object and notice the new `/sitemap-faust-pages.xml` page show up in the sitemap index
8. Go to any existing route (e.g. `/custom-page`) to ensure it doesn't get caught by the middleware
